chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/scenario/UserTerminationStrategy.java

```diff
@@ -86,6 +87,19 @@ public boolean isTerminationRequired(WorkloadExecutionStatus status) {
                 return true;
             }
         }
+    
+        boolean everyNodeHasEmptyTransactions = true;
+        for (NodeId nodeId : status.getNodes()) {
+            if (status.getEmptyTransactions(nodeId) == 0) {
+                everyNodeHasEmptyTransactions = false;
+                break;
+            }
+        }
+        if (everyNodeHasEmptyTransactions) {
+            log.info("Request to terminate work. Every node has empty transactions which means there are nothing to do any more");
```

Comment:
we are not supposed to know the exact reason of empty transaction here. it is kinda implementation specific